upf/transformers/conditional_effects_remover.py

```diff
@@ -65,19 +64,20 @@ def get_rewritten_problem(self) -> Problem:
                         nv = self._simplifier.simplify(em.Or(em.And(c, v), em.And(em.Not(c), f)))
                         self._new_problem.add_timed_effect(t, f, nv)
                 else:
-                    self._new_problem.add_timed_effect(t, e)
-        self._new_problem_add_timed_goals()
-        self._new_problem_add_maintain_goals()
-        self._new_problem_add_goals()
+                    self._new_problem._add_effect_instance(t, e)
+        self._new_problem.clear_actions()
         for ua in self._problem.unconditional_actions():
             self._new_problem.add_action(ua)
+            self._new_to_old[ua] = ua
+            self._map_old_to_new_action(ua, ua)
         for action in self._problem.conditional_actions():
-            self._counter = 0
+            self._count = 0
```

Comment:
This attribute should be hidden inside the base class

Reply:
The method `_get_fresh_action_name` should work without the need of calling other methods

Reply:
Yes, it works, but I like it better that the counter is reset for every different action. So in a problem with a very high number of actions that create a huge number of actions you don't end with super big `_count` for no reason.

It works in both ways, it's just a readability choiche

Reply:
You can always start with `count = 0` inside the `_get_fresh_action_name` and you will have the same behavior.

Reply:
Actually no because the check for the fresh name is done on the original problem. We could simply shift it onto the new problem and then you will be right, and it will also be a better naming. (Because now, not every action created is added, so you could have an action name "move", splitted into "move_0", "move_1", "move_2" and "move_3", but only "move_0" and "move_3" actually added to the problem. So the readability also might become misleading.

While instead, if we check onto the new problem, if the user gives appropriate names, it will have an easier readability where every action created being in order without missing numbers.